gk/main.c

```diff
@@ -542,6 +604,42 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 	return ret;
 }
 
+static int
+drop_flow_entry_randomly(
+	struct gk_instance *instance, unsigned int flow_ht_size)
+{
+	int  ret;
+	uint32_t entry_id = rte_rand() % flow_ht_size;
+	uint32_t entry_start_idx = entry_id;
+	struct flow_entry *fe;
+
+	while (true) {
+		fe = &instance->ip_flow_entry_table[entry_id];
+		if (fe->last_active_ts == 0) {
+			entry_id = (entry_id + 1) % flow_ht_size;
+			fe = NULL;
+		} else
+			break;
+
+		if (entry_start_idx == entry_id)
+			break;
+	}
+
+	if (fe == NULL)
+		return -1;
+
+	ret = rte_hash_del_key(instance->ip_flow_hash_table, &fe->flow);
+	if (likely(ret >= 0))
+		memset(fe, 0, sizeof(*fe));
+	else {
+		RTE_LOG(ERR, HASH,
+			"gk: failed to call rte_hash_del_key() at %s - %s",
+			__func__, strerror(-ret));
```

Comment:
Is there a reason why the error logging here for this function is different than when we called it in gl_flow_tbl_scan()?